# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/sagarjana00/FlyRank_AI_ML_Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My lane

I am choosing the **Refresh / Content Opportunity Scoring** lane as my provisional lane.

I chose this lane because the starter dataset contains content-level search and performance signals that can potentially be used to identify pages that deserve human review.

My initial goal is to investigate whether these observable signals can help prioritize pages for refresh, expansion, protection, pruning, or monitoring. This is a provisional choice and may change as I explore the data further.


In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Which content pages should be prioritized for human review
based on observable search and engagement signals?
The unit of analysis is a content page.


The decision is which pages should be reviewe## The question

### Research question

Which content pages should be prioritized for human review based on observable search and engagement signals?

### Decision

The decision is which content pages should be reviewed first for possible refresh, expansion, protection, pruning, or monitoring.

### Who acts on the decision?

A content or SEO team would use the prioritized list to decide which pages to investigate first.

### Action

After reviewing the evidence for a prioritized page, the team could decide to refresh, expand, protect, prune, or monitor the page.

### Cost of a wrong recommendation

A false positive could cause the team to spend limited time reviewing a page that does not have a meaningful opportunity.

A false negative could cause a page with a genuine performance issue or opportunity to be overlooked.

The goal is therefore to support prioritization and human review rather than automatically deciding what should be done to a page or guaranteeing that an action will improve its performance.
d first.

The possible action could be to refresh, expand, protect,
prune, or monitor a page depending on the evidence.


A false positive could cause the team to spend limited review
time on a page that does not have a meaningful opportunity.

A false negative could cause an important page with a genuine
performance problem to be overlooked.

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Shape:", df.shape)
display(df.head())

Shape: (30000, 44)


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [13]:
print(df.columns.tolist())

['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [14]:
print("Number of rows:", len(df))
print("\nTrend distribution:")
print(df["trend_direction"].value_counts())

Number of rows: 30000

Trend distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


In [15]:
print("Total rows:", len(df))
print("Unique content IDs:", df["content_id"].nunique())
print("Unique client IDs:", df["client_id"].nunique())

Total rows: 30000
Unique content IDs: 30000
Unique client IDs: 32


In [16]:
print("\nImpressions statistics:")
print(df["impressions_90d"].describe())


Impressions statistics:
count     30000.000000
mean       5200.366300
std       16838.019547
min           1.000000
25%          81.000000
50%         731.000000
75%        3615.250000
max      517715.000000
Name: impressions_90d, dtype: float64


## Quick look at the data

The starter dataset contains **30,000 observations and 30,000 unique content IDs**, so each observation corresponds to a unique content item in this starter dataset.

There are **32 unique clients** represented in the dataset.

The `trend_direction` column contains **16,262 observations classified as `down`, which is approximately 54.21% of the dataset**.

The median number of 90-day impressions is **731**, while the mean is approximately **5,200**. The large difference between the median and mean suggests that impression volume is unevenly distributed across content items.

These initial numbers make the refresh/content opportunity problem worth investigating further. However, the downward-trend label by itself does not establish that a page should be refreshed or that a refresh would improve its performance.


In [17]:
down_count = (df["trend_direction"] == "down").sum()
down_percentage = down_count / len(df) * 100

print(f"Pages with downward trend: {down_count:,}")
print(f"Percentage with downward trend: {down_percentage:.2f}%")

Pages with downward trend: 16,262
Percentage with downward trend: 54.21%


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What I can claim

I can investigate whether observable search, traffic, engagement, freshness, and content signals are associated with content performance.

I can use these signals to develop a ranking or prioritization approach for human review.

I can report patterns observed in the starter dataset and test whether a proposed scoring or modeling approach performs better than a simple baseline.

## What I cannot claim

I cannot claim that a page will definitely improve after being refreshed.

I cannot claim that the observed relationships prove causation.

I cannot claim that the identified signals are factors used by Google's ranking algorithm.

I also cannot treat the current `trend_direction` value as proof that a page is genuinely declining, because changes in performance can also have other explanations such as consolidation, seasonality, search-result changes, or noise.

Stronger conclusions will require further analysis and validation.


In [18]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.